packages/flutter/lib/src/widgets/scroll_metrics.dart

```diff
@@ -105,11 +121,22 @@ class FixedScrollMetrics extends ScrollMetrics {
   });
 
   /// Creates an immutable snapshot of the given metrics.
-  FixedScrollMetrics.clone(ScrollMetrics parent) :
-    minScrollExtent = parent.minScrollExtent,
-    maxScrollExtent = parent.maxScrollExtent,
-    pixels = parent.pixels,
-    viewportDimension = parent.viewportDimension,
+  ///
+  /// The named arguments allow the values to be optionally adjusted in the
+  /// process. See also [cloneMetrics].
+  ///
+  /// Values that are null in the given `parent` will have any specified deltas
+  /// ignored (the value remains null in the clone).
+  FixedScrollMetrics.clone(ScrollMetrics parent, {
+    double deltaMinScrollExtent: 0.0,
```

Comment:
In retrospect, I guess letting the null-ness of the fields be directly inferred from the null-ness of the constructor arguments is clearer than the null-ness of one of the fields of the parent (base on chat about copyWith)

Reply:
I'll submit a patch that does the copyWith thing instead.